src/Plugin/Sharding/Payload.php

```diff
@@ -58,14 +58,38 @@ public static function getInfo(): string {
 	 * @return static
 	 */
 	public static function fromRequest(Request $request): static {
-		$type = strtolower(strtok(trim($request->payload), ' ') ?: '');
-		return match ($type) {
+		return match ($request->command) {
 			'create', 'alter' => static::fromCreate($request),
 			'drop' => static::fromDrop($request),
+			'desc', 'show' => static::fromDesc($request),
 			default => throw new QueryParseError('Failed to parse query'),
 		};
 	}
 
+	/**
+	 * @param Request $request
+	 * @return static
+	 * @throws QueryParseError
+	 */
+	protected static function fromDesc(Request $request): static {
+		$pattern = '/(?:DESC|SHOW\s+CREATE\s+TABLE)\s+(?P<table>[^:\s\()]+)/ius';
```

Comment:
It can be DESCRIBE too - https://manual.manticoresearch.com/Listing_tables#DESCRIBE